### DISCUSSION TITLE
CalenderItem getting User not Subject, Body is 'None'

### DISCUSSION THREAD

**Discussion by Lothrien** [ecederstrand/exchangelib#1346] [Q&A] [answered]
Having a little problem with a scenario to display meetings.

The Exchange 2019 is sending <name> instead of <subject> and <body> is 'None'.

I have tried to set permission in exchange to an advanced level instead of {AvailabilityOnly}, but that doesn't solve it. Setting Folder with -DetailLevel FullDetails didn't work, because the OWA hasn't an outside address (NoExternalOwaAvailableException).

```
test_account = Account(
    primary_smtp_address = room_email,
    config = config,
    autodiscover = False,
    access_type = DELEGATE
    #access_type = IMPERSONATION # not working, change not applicated
)
now = datetime.now()
now = now.replace(tzinfo=tzone)
start_date = now
hours_to_next_day = how_many_hours_until_midnight()
end_date = now + timedelta(hours=hours_to_next_day)

ews_bfr = EWSDateTime.from_datetime(start_date)
ews_bfr_end = EWSDateTime.from_datetime(end_date)

items = test_account.calendar.view(start=ews_bfr, end=ews_bfr_end)
for item in items:
    print(item.start, item.end, item.subject, item.body, item.location, item.organizer)
```

**Comment by ecederstrand** [comment]
Can you describe your n more detail what you are trying to achieve? Also, are you trying to emulate something with exchangelib that you can do with Outlook already?

**Reply by ecederstrand** [reply]
If you're asking why the server is sending unexpected values for the subject and body fields, I think it's better to ask in a forum specific to Exchange Server. exchangelib just displays whatever the server sends.

**Reply by Lothrien** [reply, marked as answer]
ok, in outlooks "meeting calender" it shows up as a meeting having my name... and the body is empty and read-only. In my calender it is correct assigned. Renaming it gives the correct name as answer. so the problem is somewhere in exchange.

Solution:
Set-CalendarProcessing <Mailbox Name> -AddOrganizerToSubject $false -DeleteSubject $false

**Comment by Lothrien** [comment]
to get impersonate access to an account set
New-ManagementRoleAssignment -Name impersonationAssignmentName -Role ApplicationImpersonation -User <Account>
